query/src/sql/planner/binder/select.rs

```diff
@@ -193,4 +213,122 @@ impl<'a> Binder {
         let new_expr = SExpr::create_unary(filter_plan.into(), child);
         Ok(new_expr)
     }
+
+    pub(super) async fn bind_set_operator(
+        &mut self,
+        bind_context: &BindContext,
+        left: &'_ SetExpr<'_>,
+        right: &'_ SetExpr<'_>,
+        op: &SetOperator,
+        all: &bool,
+    ) -> Result<(SExpr, BindContext)> {
+        let (left_expr, left_bind_context) = self.bind_set_expr(bind_context, &*left, &[]).await?;
+        let (right_expr, right_bind_context) =
+            self.bind_set_expr(bind_context, &*right, &[]).await?;
+        if left_bind_context.columns.len() != right_bind_context.columns.len() {
+            return Err(ErrorCode::SemanticError(
+                "SetOperation must have the same number of columns",
+            ));
+        } else {
+            for (left_col, right_col) in left_bind_context
+                .columns
+                .iter()
+                .zip(right_bind_context.columns.iter())
+            {
+                if !left_col.data_type.eq(&right_col.data_type) {
+                    return Err(ErrorCode::SemanticError(
+                        "SetOperation's types cannot be matched",
+                    ));
+                }
+            }
+        }
+        match (op, all) {
+            (SetOperator::Intersect, false) => {
+                // Transfer Intersect to Semi join
+                self.bind_intersect(left_bind_context, right_bind_context, left_expr, right_expr)
+            }
+            (SetOperator::Except, false) => {
+                // Transfer Except to Anti join
+                self.bind_except(left_bind_context, right_bind_context, left_expr, right_expr)
+            }
+            _ => Err(ErrorCode::UnImplement("Unsupported query type, currently, databend only support intersect distinct and except distinct")),
+        }
+    }
+
+    fn bind_intersect(
+        &mut self,
+        left_context: BindContext,
+        right_context: BindContext,
+        left_expr: SExpr,
+        right_expr: SExpr,
+    ) -> Result<(SExpr, BindContext)> {
+        self.bind_intersect_or_except(
+            left_context,
+            right_context,
+            left_expr,
+            right_expr,
+            JoinType::SemiJoin,
+        )
+    }
+
+    fn bind_except(
+        &mut self,
+        left_context: BindContext,
+        right_context: BindContext,
+        left_expr: SExpr,
+        right_expr: SExpr,
+    ) -> Result<(SExpr, BindContext)> {
+        self.bind_intersect_or_except(
+            left_context,
+            right_context,
+            left_expr,
+            right_expr,
+            JoinType::AntiJoin,
+        )
+    }
+
+    fn bind_intersect_or_except(
+        &mut self,
+        left_context: BindContext,
+        right_context: BindContext,
+        left_expr: SExpr,
+        right_expr: SExpr,
+        join_type: JoinType,
+    ) -> Result<(SExpr, BindContext)> {
+        let left_expr = self.bind_distinct(
```

Comment:
Better not to reuse the `bind_distinct` here, since it's not actually a `DISTINCT` clause. Construct an `Aggregate` operator here is fine.

Reply:
Yes, I knew `bind_distinct` just wrapped a `Aggregate` plan for current `SExpr`.

Here my aim is to remove duplicates for `left_expr` to implement `intersect` or `except` **distinct**, so I think it's ok to reuse `bind_distinct`, which fits the context.